src/main/java/org/folio/circulation/domain/notice/schedule/ScheduledDigitalReminderHandler.java

```diff
@@ -97,21 +112,48 @@ protected CompletableFuture<Result<ScheduledNoticeContext>> fetchLoan(
       .thenCompose(r -> r.after(loanPolicyRepository::findPolicyForLoan))
       .thenCompose(overdueFinePolicyRepository::findOverdueFinePolicyForLoan)
       .thenApply(mapResult(context::withLoan))
+      .thenApply(r -> {
+        String servicePointId = r.value().getLoan().getCheckoutServicePointId();
+        calendarRepository.lookupOpeningDays(ClockUtil.getZonedDateTime().toLocalDate(),servicePointId);
+        return r;
+      })
       .thenApply(r -> r.next(this::failWhenLoanIsIncomplete));
   }
-
+//
   @Override
   protected CompletableFuture<Result<ScheduledNoticeContext>> fetchData(ScheduledNoticeContext context) {
     return ofAsync(() -> context)
       .thenApply(r -> r.next(this::failWhenNoticeHasNoLoanId))
       .thenCompose(r -> r.after(this::fetchTemplate))
       .thenCompose(r -> r.after(this::fetchLoan))
       .thenApply(r -> r.next(this::failWhenLoanHasNoNextReminderScheduled))
-      .thenCompose(r -> r.after(this::buildAccountObject));
+      .thenCompose(r -> r.after(this::instantiateReminderFeeAccount));
+  }
+
+  private CompletableFuture<Result<Boolean>> isOpenDay(ScheduledNoticeContext noticeContext) {
+    String servicePointId = noticeContext.getLoan().getCheckoutServicePointId();
+    return getSystemTimeInTenantsZone()
+      .thenCompose(tenantTime ->
+        calendarRepository.lookupOpeningDays(tenantTime.toLocalDate(),servicePointId)
+          .thenCompose(days -> {
+            Boolean openDay = days.value().getRequestedDay().isOpen();
+            return ofAsync(openDay);
```

Comment:
Is it possible to use Result methods instead (like after(), next() etc.) instead? Most of the time, when you have to call Result.value() that's an indication that something was done wrong.